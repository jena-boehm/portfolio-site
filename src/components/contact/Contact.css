.contactContainer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: column;
    height: max-content;
    width: 100%;
    box-sizing: border-box;
    color: white;

    .titleLeft {
        transform: rotate(270deg);
        font-size: 7.3em;
        letter-spacing: 0.05em;
        margin: 0;
        padding: 0;
        height: max-content;
        opacity: 0.3;
        position: fixed;
        left: -13vw;
        top: 46vh;
        font-weight: 300;
        color: #B0B6FB;
        text-shadow: 6px 6px 1px black;
    }

    .aboutRight {
        display: flex;
        justify-content: center;
        align-items: center;
        flex-direction: column;
        height: 100%;
        width: 90%;
        margin: 0 2em 1em 10em;
        padding: 0;

        .rightMain {
            width: 60%;
            height: max-content;
            display: flex;
            justify-content: center;
            align-items: center;
            flex-direction: column;
            margin: 0.5em 0 1em 0;

            .rightTop {
                width: 100%;
                display: flex;
                justify-content: center;
                align-items: center;
                flex-direction: column;
                margin: 1em 0 1em 0;
                font-size: 2em;
            }

            .form {
                width: 80%;
                display: flex;
                justify-content: center;
                align-items: center;
                flex-direction: column;
                border: 1px solid rgba(176, 182, 251, 0.2);
                box-shadow: rgba(176, 182, 251, 0.3) 0px 2px 5px 1px, rgba(176, 182, 251, 0.23) 0px 2px 6px 1px;
                padding: 2em 0 1.5em 0;

                .formInput { 
                    position: relative;
                    width: 70%;

                    label { 
                        position: absolute;
                        top: 0;
                        left: 0;
                        padding: 10px 0;
                        font-size: 1em;;
                        color: #fff;
                        pointer-events: none;
                        transition: .5s;
                    }

                    input {
                        color: black;
                        width: 100%;
                        padding: 10px 0;
                        font-size: 16px;
                        color: #fff;
                        margin-bottom: 30px;
                        border: none;
                        border-bottom: 1px solid #fff;
                        outline: none;
                        background: transparent;
                    }

                    input:focus + label {
                        top: -20px;
                        left: 0;
                        color: #B0B6FB;
                        font-size: 12px;
                    }

                    input:valid + label {
                        top: -20px;
                        left: 0;
                        color: #B0B6FB;
                        font-size: 12px;
                    }
                }

                button { 
                    border: none;
                    outline: none;
                    background: transparent;
                    text-transform: uppercase;
                    box-shadow: 0 0 2px 0px #B0B6FB;
                    padding: 0.5em 1em 0.5em 1em;
                    letter-spacing: 0.1em;
                    cursor: pointer;
                    font-size: 1em;
                    color: #B0B6FB;
                    transition: 0.2s;
                }

                button:hover {
                        color: white;
                        font-size: 1.2em;
                        padding: 0.3em 1.1em 0.3em 1.1em;
                        transition: 0.2s;
                }

                button:active {
                    font-size: 1em;
                     box-shadow: inset 0 0 2px 0px #B0B6FB;
                     padding: 0.5em 1em 0.5em 1em;

                }
            }

            .thankYou {
                text-align: center;
                margin: 5em 0 0 2em;
                font-size: 1.5em;
                opacity: 0.8;
            }
        }

        .socials {
            display: flex;
            justify-content: center;
            align-items: center;
            margin-top: 1em;
            opacity: 0.8;
        }
    }
}