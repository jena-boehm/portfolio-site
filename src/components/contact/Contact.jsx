/* eslint-disable max-len */
import React, { useState, useEffect } from 'react';
import styles from './Contact.css';
import { Socials } from '../socials/Socials';

export const Contact = () => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [role, setRole] = useState('');
  const [company, setCompany] = useState('');
  const [message, setMessage] = useState('');
  const [isSent, setIsSent] = useState(false);
  const [isMobile, setMobile] = useState(window.innerWidth < 1100);

  const handleMobile = () => {
    setMobile(window.innerWidth < 1100);
  };

  useEffect(() => {
    window.addEventListener('resize', handleMobile);
    return () => window.removeEventListener('resize', handleMobile);
  });

  const handleSubmit = e => {
    e.preventDefault();
    // fetch(`${process.env.ZAPIER_URL}`, {
    //   method: 'POST',
    //   body: JSON.stringify({ name, email, role, company, message })
    // })
    setIsSent(true)
      .catch(() => alert('There was an error, please try again'));
  };

  const thankYouMessage = <div className={styles.thankYou}>Thanks for reaching out! We&apos;ll be in touch. <br/>In the meantime, let&apos;s connect on LinkedIn and GitHub!</div>;

  const contactForm = <>
    <section className={styles.rightTop}>
      Let&apos;s chat!
    </section>  
    <form className={styles.form} name="portfolio-contact" autoComplete="none" data-netlify="true" onSubmit={handleSubmit}>

      <div className={styles.formInput}>
        <input
          required
          type="portfolioName7890"
          name="portfolioName"
          autoComplete="none"
          onChange={e => setName(e.target.value)}/>
        <label htmlFor="contactName7890">Name</label>
      </div>
    
      <div className={styles.formInput}>
        <input
          required
          type="portfolioEmail7890"
          name="portfolioEmail7890"
          autoComplete="none"
          onChange={e => setEmail(e.target.value)}/>
        <label htmlFor="contactEmail7890">Email</label> 
      </div>
      
      <div className={styles.formInput}>
        <input
          required
          type="portfolioRole7890"
          name="portfolioRole7890"
          autoComplete="none"
          onChange={e => setRole(e.target.value)}/>  
        <label htmlFor="contactRole7890">Role</label> 
      </div>

      <div className={styles.formInput}>
        <input
          required
          type="portfolioCompany7890"
          name="portfolioCompany7890"
          autoComplete="none"
          onChange={e => setCompany(e.target.value)}/>
        <label htmlFor="contactCompany7890">Company</label> 
      </div>

      <div className={styles.formInput}>
        <input
          required
          type="portfolioMessage7890"
          name="portfolioMessage7890"
          autoComplete="on"
          onChange={e => setMessage(e.target.value)}/>
        <label htmlFor="contactMessage7890">Message</label> 
      </div>

      <button type="submit">Send</button>
    </form>
  </>;

  return (
    <div className={styles.contactContainer}>
      <h1 className={styles.titleLeft}>
            CONTACT
      </h1>
      <section className={styles.contactRight}>
        <section className={styles.rightMain}>
          {isMobile 
        && <div className={styles.titleTop}>CONTACT</div>}
          {isSent ? thankYouMessage : contactForm}
        </section>
        <section className={styles.socials}>
          <Socials />
        </section>
      </section>
    </div>
  );
};
