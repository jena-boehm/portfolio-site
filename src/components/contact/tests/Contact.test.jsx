import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Contact } from '../Contact';

describe('Contact component', () => {
  afterEach(() => cleanup());
  it('renders Contact', () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });
});
