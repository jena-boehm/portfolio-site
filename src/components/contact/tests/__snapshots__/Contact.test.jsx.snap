// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders Contact 1`] = `
<DocumentFragment>
  <div
    class="contactContainer"
  >
    <h1
      class="titleLeft"
    >
      CONTACT
    </h1>
    <section
      class="aboutRight"
    >
      <section
        class="rightMain"
      >
        <section
          class="rightTop"
        >
          Let's chat!
        </section>
        <form
          autocomplete="none"
          class="form"
        >
          <div
            class="formInput"
          >
            <input
              autocomplete="none"
              name="contactName7890"
              required=""
              type="contactName7890"
            />
            <label
              for="contactName7890"
            >
              Name
            </label>
          </div>
          <div
            class="formInput"
          >
            <input
              autocomplete="none"
              name="contactEmail7890"
              required=""
              type="contactEmail7890"
            />
            <label
              for="contactEmail7890"
            >
              Email
            </label>
          </div>
          <div
            class="formInput"
          >
            <input
              autocomplete="none"
              name="contactRole7890"
              required=""
              type="contactRole7890"
            />
            <label
              for="contactRole7890"
            >
              Role
            </label>
          </div>
          <div
            class="formInput"
          >
            <input
              autocomplete="none"
              name="contactCompany7890"
              required=""
              type="contactCompany7890"
            />
            <label
              for="contactCompany7890"
            >
              Company
            </label>
          </div>
          <div
            class="formInput"
          >
            <input
              autocomplete="on"
              name="contactMessage7890"
              required=""
              type="contactMessage7890"
            />
            <label
              for="contactMessage7890"
            >
              Message
            </label>
          </div>
          <button
            type="submit"
          >
            Send
          </button>
        </form>
      </section>
      <section
        class="socials"
      >
        <div
          class="socialsContainer"
        >
          <div
            class="icons"
          >
            <div
              class="iconContainer"
            >
              <a
                href="https://www.linkedin.com/in/jenaluxboehm/"
              >
                <img
                  alt="LinkedIn Icon"
                  class="iconBottom"
                  src="/path/image"
                />
                <div
                  class="iconTop"
                >
                  <img
                    alt="LinkedIn Icon"
                    class="imgTop"
                    src="/path/image"
                  />
                </div>
              </a>
            </div>
            <div
              class="iconContainer"
            >
              <a
                href="https://github.com/jena-boehm"
              >
                <img
                  alt="GitHub Icon"
                  class="iconBottom"
                  src="/path/image"
                />
                <div
                  class="iconTop"
                >
                  <img
                    alt="GitHub Icon"
                    class="imgTop"
                    src="/path/image"
                  />
                </div>
              </a>
            </div>
            <div
              class="iconContainer"
            >
              <a
                href="mailto:jenaluxboehm@gmail.com"
              >
                <img
                  alt="Email Icon"
                  class="iconBottom"
                  src="/path/image"
                />
                <div
                  class="iconTop"
                >
                  <img
                    alt="Email Icon"
                    class="imgTop"
                    src="/path/image"
                  />
                </div>
              </a>
            </div>
          </div>
          <button>
            Resume
          </button>
        </div>
      </section>
    </section>
  </div>
</DocumentFragment>
`;
